### PR TITLE
Make chantal clone only needed branch

### PR DESCRIPTION
**Pull request by fabiobarkoski**
changed the default --depth from 0 to 5, so that way chantal will only clone one branch implicitly with the specified number of commits

### CODE REVIEW

File: chantal/build.py [hunk 1]
```diff
@@ -41,18 +41,23 @@ def build_job(args):
     if args.clone_location:
         job_state("running", "cloning repo")
 
+        if args.treeish:
+            branch = ("--branch %s " % args.treeish)
```
Comment (TheJJ) [inline chantal/build.py:45]: we use treeish for exact commit hashes to check out, and --branch will only accept branches/tags, right?

**Review by TheJJ — CHANGES_REQUESTED** [review]
Thanks! Can we keep the default of 0 (full clone) for now please?
The clone depth should be configurable in the project settings, then each user can configure the desired value easily.

File: chantal/build.py (hunk 1 above)
Comment (fabiobarkoski) [reply chantal/build.py:45]: Yeah, I already changed this, and I'm testing these changes with the `depth` on project settings. Soon as possible, I'll update the draft, sorry for the slow implementation.

Comment (TheJJ) [reply chantal/build.py:45]: haha no worries :) you're supposed to have fun, so take your time!

**Comment by fabiobarkoski** [comment]
Is there any possibility of the `treeish` receive a branch instead of a commit hash? Because the [depth](https://git-scm.com/docs/git-clone#Documentation/git-clone.txt---depthltdepthgt) flag implicit use `--single-branch` on cloning, so chantal will can't change the branch unless we use `--no-single-branch`

File: etc/project.conf.example [hunk 2]
```diff
@@ -16,6 +16,8 @@ job_max_output = 100MiB
 # name of the build job task configuration file within the repo
 job_desc_file = kevinfile
 
+# git configuration
+depth_commits = 0
```
Comment (TheJJ) [inline etc/project.conf.example:20]: perhaps `git_clone_depth` is a more speaking name for this?

File: kevin/chantal.py [hunk 3]
```diff
@@ -193,6 +194,7 @@ def run(self, job):
             ("python3", "-u", "-m", "chantal",
              "--clone", job.build.clone_url,
              "--checkout", job.build.commit_hash,
+             "--depth", self.git_config["shallow"],
```
Comment (TheJJ) [inline kevin/chantal.py:197]: the option preset in chantal is called `--clone_depth`

**Review by TheJJ — COMMENTED** [review]
the github ci executes the following commands:

```
config:
  repository: stustapay/stustapay
  token: ***
  ssh-strict: true
  persist-credentials: true
  clean: true
  fetch-depth: 1
  lfs: false
  submodules: false
  set-safe-directory: true
  ```
 
 ```
/usr/bin/git init /home/runner/work/stustapay/stustapay
/usr/bin/git remote add origin https://github.com/stustapay/stustapay
/usr/bin/git config --local http.https://github.com/.extraheader AUTHORIZATION: basic $secret

/usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +05317cdc9f8cb692bec0e75bc1d573eed86f48a4:refs/remotes/pull/260/merge

/usr/bin/git checkout --progress --force refs/remotes/pull/260/merge

/usr/bin/git log -1 --format='%H'
```

maybe this approach instead of git clone is also better for us - we can get any treeish of any depth then :)
if you are interested, we could just switch our clone call with (a variant) of those commands.

**Comment by fabiobarkoski** [comment]
> the github ci executes the following commands:
> 
> ```
> config:
>   repository: stustapay/stustapay
>   token: ***
>   ssh-strict: true
>   persist-credentials: true
>   clean: true
>   fetch-depth: 1
>   lfs: false
>   submodules: false
>   set-safe-directory: true
> ```
> 
> ```
> /usr/bin/git init /home/runner/work/stustapay/stustapay
> /usr/bin/git remote add origin https://github.com/stustapay/stustapay
> /usr/bin/git config --local http.https://github.com/.extraheader AUTHORIZATION: basic $secret
> 
> /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +05317cdc9f8cb692bec0e75bc1d573eed86f48a4:refs/remotes/pull/260/merge
> 
> /usr/bin/git checkout --progress --force refs/remotes/pull/260/merge
> 
> /usr/bin/git log -1 --format='%H'
> ```
> 
> maybe this approach instead of git clone is also better for us - we can get any treeish of any depth then :) if you are interested, we could just switch our clone call with (a variant) of those commands.

Sure :)

File: kevin/chantal.py (hunk 3 above)
Comment (fabiobarkoski) [reply kevin/chantal.py:197]: Oh, true

File: etc/project.conf.example (hunk 2 above)
Comment (fabiobarkoski) [reply etc/project.conf.example:20]: Yeah, it's good, but if I use fetch instead clone, `git_fetch_depth` is a good one?

File: chantal/build.py [hunk 4]
```diff
@@ -46,13 +46,16 @@ def build_job(args):
...
             shallow = ""
 
-        run_command("git clone " + shallow +
-                    shlex.quote(args.clone_location) +
-                    " " + args.work_location, base_env)
+        run_command("git init " + args.work_location, base_env)
 
     os.chdir(args.work_location)
 
     if args.treeish:
```
Comment (TheJJ) [inline chantal/build.py:53]: if we don't get a treeish we would not fetch any repo content - so the git init should be part of this condition, and in the else part we fall back to git clone

**Review by TheJJ — COMMENTED** [review]
great, nearly there :)

**Review by TheJJ — APPROVED** [review]

File: chantal/build.py [hunk 5]
```diff
@@ -46,13 +46,21 @@ def build_job(args):
...
+        else:
+            run_command("git clone " + shallow +
+                        shlex.quote(args.clone_location) +
+                        " " + args.work_location, base_env)
 
     os.chdir(args.work_location)
 
     if args.treeish:
+        run_command("git remote add origin " +
+                    shlex.quote(args.clone_location), base_env)
```
Comment (TheJJ) [inline chantal/build.py:60]: i think this will fail if we have cloned a repo, since then the remote named origin is already there.

**Review by TheJJ — CHANGES_REQUESTED** [review]
we need to merge both if-treeish and the chdir so when having a treeish we do the whole init-chdir-addorigin-fetch-checkout flow and the cloning-chdir separately

File: chantal/build.py (hunk 5 above)
Comment (fabiobarkoski) [reply chantal/build.py]: `clone` and `add origin` will run in different cases. `clone` will only exec if there's no treeish and if so, `add origin` will.

**Comment by fabiobarkoski** [comment]
> we need to merge both if-treeish and the chdir so when having a treeish we do the whole init-chdir-addorigin-fetch-checkout flow and the cloning-chdir separately

Something like this?
https://github.com/SFTtech/kevin/blob/a56fcec3a7eef00c3f84dd7d23f61e104a5eaef6/chantal/build.py#L49-L62

**Review by TheJJ — APPROVED** [review]

**Comment by TheJJ** [comment]
great! by the way, do you use kevin for some of your projects?

**Comment by fabiobarkoski** [comment]
> great! by the way, do you use kevin for some of your projects?

No, just contributing for openage :)

**Pull request merged**